src/providers/NearWalletProvider.tsx

```diff
@@ -0,0 +1,160 @@
+"use client"
+
+import { NearConnector } from "@hot-labs/near-connect"
+import type { FinalExecutionOutcome } from "@near-wallet-selector/core"
+import type {
+  SignMessageParams,
+  SignedMessage,
+} from "@near-wallet-selector/core/src/lib/wallet/wallet.types"
+import { FeatureFlagsContext } from "@src/providers/FeatureFlagsProvider"
+import type { SignAndSendTransactionsParams } from "@src/types/interfaces"
+import { logger } from "@src/utils/logger"
+import { getDomainMetadataParams } from "@src/utils/whitelabelDomainMetadata"
+import {
+  type FC,
+  type ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react"
+
+interface NearWalletContextValue {
+  connector: NearConnector | null
+  accountId: string | null
+  connect: () => Promise<void>
+  disconnect: () => Promise<void>
+  signMessage: (message: SignMessageParams) => Promise<{
+    signatureData: SignedMessage
+    signedData: SignMessageParams
+  }>
+  signAndSendTransactions: (
+    params: SignAndSendTransactionsParams
+  ) => Promise<FinalExecutionOutcome[]>
+}
+
+export const NearWalletContext = createContext<NearWalletContextValue | null>(
+  null
+)
+
+export const NearWalletProvider: FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [connector, setConnector] = useState<NearConnector | null>(null)
+  const [accountId, setAccountId] = useState<string | null>(null)
+  const { whitelabelTemplate } = useContext(FeatureFlagsContext)
+
+  const init = useCallback(async () => {
+    const _connector = new NearConnector({
+      network: "mainnet",
+      walletConnect: {
+        ...getDomainMetadataParams(whitelabelTemplate),
+      },
```

Comment:
maybe this would be enough
```suggestion
      walletConnect: getDomainMetadataParams(whitelabelTemplate),
```